core/Network.js

```diff
@@ -86,12 +87,20 @@ Network.prototype.getProcessByName = function (processName) {
 
 Network.prototype.run = function (runtime, options, callback) {
   options = options || {};
-
-  _.forEach(this._processes, function (process) {
-    _.invokeMap(process.inports, 'setRuntime', runtime);
-    _.invokeMap(process.outports, 'setRuntime', runtime);
-  });
-  runtime.run(_.values(this._processes), options, callback || function () {});
+  callback = callback || function () {};
+
+  _.invokeMap(this._connections, 'setRuntime', runtime);
+
+  try {
+    runtime.run(_.values(this._processes), options, callback );
+  } catch (e) {
+    console.log('Connections');
```

Comment:
I am finding this very useful. When there's a deadlock, show all the connections and which ones are full. This may help identify *why* there's a deadlock.